apps/chat/src/components/Common/ApplicationWizard/form.ts

```diff
@@ -0,0 +1,193 @@
+import { Validate } from 'react-hook-form';
+
+import {
+  getModelDescription,
+  getQuickAppConfig,
+} from '@/src/utils/app/application';
+import { notAllowedSymbols } from '@/src/utils/app/file';
+
+import { CustomApplicationModel } from '@/src/types/applications';
+import { DialAIEntityFeatures } from '@/src/types/models';
+import { QuickAppConfig } from '@/src/types/quick-apps';
+
+import { DEFAULT_TEMPERATURE } from '@/src/constants/default-ui-settings';
+
+import isObject from 'lodash-es/isObject';
+
+export interface FormData {
+  name: string;
+  description: string;
+  version: string;
+  iconUrl: string;
+  topics: string[];
+  inputAttachmentTypes: string[];
+  maxInputAttachments: number | undefined;
+  completionUrl: string;
+  features: string | null;
+  instructions: string;
+  temperature: number;
+  toolset: string;
+}
+
+type Validator = Partial<{
+  required: string;
+  validate: Validate<FormData[keyof FormData], Partial<FormData>>;
+}>;
+
+export const validators: Partial<Record<keyof FormData, Validator>> = {
+  name: {
+    required: 'This field is required',
+    validate: (v) => {
+      const reg = new RegExp(`^[^${notAllowedSymbols}]{2,160}$`);
+
+      return (
+        reg.test(v as string) ||
+        'Name should be 2 to 160 characters long and should not contain special characters'
+      );
+    },
+  },
+  version: {
+    required: 'This field is required',
+    validate: (v) => {
+      const reg = /^[0-9]+\.[0-9]+\.[0-9]+$/;
+
+      return (
+        reg.test(v as string) ||
+        'Version should be in x.y.z format and contain only numbers and dots.'
+      );
+    },
+  },
+  iconUrl: {
+    required: 'Icon is required',
```

Comment:
do we have translation for errors/required messages?

Reply:
yes we do, in error component